src/controllers/orderController.js

```diff
@@ -5,58 +5,100 @@ const { BadRequestError } = require('../errors');
+
+    // Create an Order document in the database
+    const order = await Order.create({
+      user: req.user.userId,
+      orderItems,
+      subtotal,
+      tax,
+      total,
+    });
 
-  // Perform necessary checks and validations
-  if (!orderItems?.length) {
-    throw new BadRequestError(
-      'Unable to create order because the order items are missing.'
-    );
-  }
+    // Create a payment intent with Stripe
+    const paymentIntent = await stripe.paymentIntents.create({
+      amount: total * 100, // Stripe requires amount in cents
+      currency: 'usd',
+      description: `Order #${order._id}`,
+      metadata: {
+        userId: req.user.userId,
+        orderId: order._id, // key-values
+        totalQuantity: orderItems.reduce(
+          (total, item) => total + item.quantity,
+          0
+        ), // the total number of items in the order
+        totalAlbums: orderItems.length, // the total number of albums purchased in order
+      },
+    });
 
-  // Create an Order document in the database
-  const order = await Order.create({
-    user: req.user.userId,
-    orderItems,
-    subtotal,
-    tax,
-    total,
-  });
+    // Save the paymentIntent to the Order model
+    order.paymentIntentId = paymentIntent.id;
+    await order.save();
 
-  // Create a payment intent with Stripe
-  const paymentIntent = await stripe.paymentIntents.create({
-    amount: total * 100, // Stripe requires amount in cents
-    currency: 'usd',
-    description: `Order #${order._id}`,
-    metadata: {
-      userId: req.user.userId,
-      orderId: order._id, // key-values
-      totalQuantity: orderItems.reduce(
-        (total, item) => total + item.quantity,
-        0
-      ), // the total number of items in the order
-      totalAlbums: orderItems.length, // the total number of albums purchased in order
-    },
-  });
+    // Create PurchasedAlbum entries for each album in orderItems
+    for (const orderItem of orderItems) {
+      await PurchasedAlbum.create({
+        album: orderItem.album, // album ID from orderItem
+        user: req.user.userId, // user ID from the request
+      });
+    }
 
-  // Save the paymentIntent to the Order model
-  order.paymentIntentId = paymentIntent.id;
-  await order.save();
 
-  // Create PurchasedAlbum entries for each album in orderItems
-  for (const orderItem of orderItems) {
-    await PurchasedAlbum.create({
-      album: orderItem.album, // album ID from orderItem
-      user: req.user.userId, // user ID from the request
-    });
-  }
 
-  // Send the payment intent client secret and order information to the client
-  res
-    .status(StatusCodes.CREATED)
-    .json({ clientSecret: paymentIntent.client_secret, order });
+    // Start a database transaction
+    await db.startTransaction();
+
+    // Save the order to the database
+    const savedOrder = await db.saveOrder(orderData);
+
+    // Send the order completion email
+    await sendOrderCompletedEmail(user.email, user.name);
```

Comment:
Will we also pass along the order items so that those can be shown in the email?

Reply:
@akosasante yes, that was my idea, I'm just not sure yet how to write that

Reply:
Yes we can do that for sure :+1:  We can pass in `order.orderItems` here or if you need other info from the order, you could even pass along the whole `order` variable (and add a matching argument where `sendOrderCompletedEmail` is defined)

Reply:
@akosasante Akos, I was trying to do that but couldn't receive albumName and price from the populated 'album' property in OrderItemSchema- could you please take a look- the maximun I was able to reach is this   {
                "album": "64d36bd25c5dec8e599bee05",
                "quantity": 1,
                "_id": "64faef78e19c2e36bdd0db53"
            },
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/103755090/6f8b91ac-8027-4e19-9df8-20c6b2c8c4ce)
I think I need to populate 'album' in OrderItemSchema so I added this code 
![image](https://github.com/Code-the-Dream-School/dd-prac-team4-back/assets/103755090/99e9b0cb-050e-48c4-bb4d-a07917554f03)
but it didn't help